onedal/spmd/linear_model/logistic_regression.py

```diff
@@ -0,0 +1,45 @@
+# ==============================================================================
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from abc import ABC
+
+from onedal.linear_model import LogisticRegression as LogisticRegression_Batch
+
+from ..._device_offload import support_usm_ndarray
+from ...common._spmd_policy import _get_spmd_policy
+
+
+class BaseLogisticRegressionSPMD(ABC):
+    def _get_policy(self, queue, *data):
+        return _get_spmd_policy(queue)
+
+
+class LogisticRegression(BaseLogisticRegressionSPMD, LogisticRegression_Batch):
+    @support_usm_ndarray()
+    def fit(self, X, y, queue=None):
+        return super().fit(X, y, queue)
+
+    @support_usm_ndarray()
+    def predict(self, X, queue=None):
+        return super().predict(X, queue)
+
+    @support_usm_ndarray()
+    def predict_proba(self, X, queue=None):
+        return super().predict_proba(X, queue)
+
+    @support_usm_ndarray()
+    def predict_log_proba(self, X, queue=None):
+        return super().predict_log_proba(X, queue)
```

Comment:
I think we can do this on sklearnex spmd estimator side when it is introduced, because we don't have docstings in any of the estimators defined in onedal/spmd